src/nvim/ex_docmd.c

```diff
@@ -5802,6 +5802,29 @@ static void ex_delcommand(exarg_T *eap)
   }
 }
 
+/// Split a string by unescaped whitespace (space & tab), used for f-args on commands
+///
+/// @note  If no separator is found start = 0 and end = length - 1
+/// @param[in]  arg  String to split
+/// @param[in]  iter Iteration counter
+/// @param[out]  start Start of the split
+/// @param[out]  end End of the split
+/// @param[in]  length Length of the string
+/// @return  false if it's the last split (don't call again), true otherwise (call again).
+bool uc_split_args_iter(const char_u *arg, int iter, int *start, int *end, int length)
```

Comment:
What is used for normal `:commands` to split args on escaped whitespace? Is it possible to reuse that instead of creating a new function?

Reply:
Just under the split function you can see the ex function, it does a lot more it allocates and quotes every parameter, and also replaces in the whole command string I believe. So reusing is not an easy option.

Reply:
Here it's https://github.com/neovim/neovim/blob/4b0b470f237074e53853e10639f25abb5c5b04df/src/nvim/ex_docmd.c#L5831

Reply:
Ok, I like the idea of avoiding an allocation if necessary.

Can you update the doctoring for this function to compare it with `uc_split_args` so that people know why we have both?

Bonus: I wonder if it would be possible to replace `uc_split_args` with `uc_split_args_iter` and remove even more allocations? 🤔

Reply:
Just did!

About the bonus, I like this comment 😄 :

https://github.com/neovim/neovim/blob/005a7aa1670841dfdffb7559161a7d37d6c14bc4/src/nvim/ex_docmd.c#L6005-L6007